### PR TITLE
WIP: Adding example illustrating usage of optimizer weights.

### PR DESCRIPTION
**Pull request by zivy**
Need a bit of help selecting appropriate inputs for the tests of this example. The images I used for developing it are too big. Cmake has "place holders" that will break testing.

### CODE REVIEW

**Review by blowekamp — APPROVED** [review]
LGTM

**Comment by zivy** [comment]
@blowekamp, any recommendations on inputs for the testing using the existing test images? I wasn't able to identify a reasonable pair of volumes. Should I just add new data, with the minimal size that still works?

**Comment by blowekamp** [comment]
A lot of the registration test data comes from ITK's 2D illustrations, which have an explicitly transformed moving image ( see BrainProtonDensitySlice* ). It really depends on what type of image you are looking for. As currently we do not show the images, perhaps the content doesn't matter too much? The "RA" image is one 3D test image that could be transformed as needed?

File: Examples/ImageRegistrationOptimizerWeights/Documentation.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Image Registration Optimizer Weights
+====================================
+
+.. include:: registrationExamplePrefix.rst
+
+Overview
+--------
+The image registration framework allows us to disable subsets of the
+transformation parameters so that we can use a lower dimensional
+transformation. Or if we have some better prior knowledge weigh the
```
Comment (dave3d) [inline Examples/ImageRegistrationOptimizerWeights/Documentation.rst:10]: "Of if we have some better prior knowledge *we can* weigh the ..."

Not a complete sentence.

File: Examples/ImageRegistrationOptimizerWeights/Documentation.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+parameters accordingly (this is much harder to utilize to get a desired
+effect than disabling parameters).
+
+In this example, we work with a 3D rigid transformation
+using the Euler parameterization but we only allow rotation around the `z`
```
Comment (dave3d) [inline Examples/ImageRegistrationOptimizerWeights/Documentation.rst:15]: comma after parameterization

File: Examples/ImageRegistrationOptimizerWeights/Documentation.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+axis. As there is no transformation that represents this subspace of rigid
+transformations we use the `SetOptimizerWeights` method to disable
```
Comment (dave3d) [inline Examples/ImageRegistrationOptimizerWeights/Documentation.rst:17]: comma after transformations

File: Examples/ImageRegistrationOptimizerWeights/Documentation.rst (hunk 1 above)
Comment (zivy) [reply Examples/ImageRegistrationOptimizerWeights/Documentation.rst]: Modified.

File: Examples/ImageRegistrationOptimizerWeights/Documentation.rst (hunk 2 above)
Comment (zivy) [reply Examples/ImageRegistrationOptimizerWeights/Documentation.rst]: done

File: Examples/ImageRegistrationOptimizerWeights/Documentation.rst (hunk 3 above)
Comment (zivy) [reply Examples/ImageRegistrationOptimizerWeights/Documentation.rst]: done

**Pull request merged**